### PR TITLE
[chore] move kafka exporter to generated lifecycle tests

### PR DESCRIPTION
**Pull request by Frapschen**
Relates to https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/27849

### CODE REVIEW

**Review by mx-psi — DISMISSED** [review]

**Comment by atoulme** [comment]
@pavolloffay and @MovieStoreGuy please review as codeowners. I have changed the start sequence of the exporter.

File: exporter/kafkaexporter/kafka_exporter.go
Comment (MovieStoreGuy) [inline exporter/kafkaexporter/kafka_exporter.go:69]: Do you mind just explaining why the producer is created on `Start` instead of `New` ?

**Comment by MovieStoreGuy** [comment]
I am nervous about moving the creation of the client within to Start mostly because it will need some locking and components are meant to be immutable once created (to a certain degree).

File: exporter/kafkaexporter/kafka_exporter.go
Comment (atoulme) [reply exporter/kafkaexporter/kafka_exporter.go:69]: With component status reporting, the expectation also is that you'd keep trying to connect to Kafka periodically, and report errors otherwise. This would allow an operator to review the issue and use opamp fix the receiver. The way this is managed right now, the component bails out on creation, and the whole collector exits early without recourse.

**Review by MovieStoreGuy — APPROVED** [review]

**Comment by MovieStoreGuy** [comment]
Happy with the changes, I just need the conflicts resolved before we can merge :)

**Pull request merged**
